### PR TITLE
Add remote-read commands

### PR DESCRIPTION
**Pull request by simonswine**
The remote-read commands allow to investigate series through the remote-read API:

 * `remote-read export`: Export metrics remote read series into a local
 TSDB.
 * `remote-read dump`: Dump remote read series.
 * `remote-read stats`: Show statistic of remote read series.

Signed-off-by: Christian Simon <simon@swine.de>

### CODE REVIEW

File: pkg/backfill/backfill.go [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+	db, err := tsdb.OpenDBReadOnly(outputDir, nil)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		var mErr tsdb_errors.MultiError
+		mErr.Add(returnErr)
+		mErr.Add(db.Close())
+		returnErr = mErr.Err()
```
Comment (replay) [inline pkg/backfill/backfill.go:83]: that's interesting, i've never seen a return value get assigned inside a deferred function, but it makes sense 👍

File: pkg/commands/remote_read.go [hunk 2]
```diff
@@ -0,0 +1,441 @@
...
+	readTimeout time.Duration
+	tsdbPath    string
+
+	selector string
+	from     string
+	to       string
+}
+
+func (c *RemoteReadCommand) Register(app *kingpin.Application) {
+	remoteReadCmd := app.Command("remote-read", "TODO")
```
Comment (replay) [inline pkg/commands/remote_read.go:48]: I guess that `TODO` should be replaced before merging?

File: pkg/commands/remote_read.go [hunk 3]
```diff
@@ -0,0 +1,441 @@
...
+	series := i.ts[i.posSeries]
+	sample := series.Samples[i.posSample]
+
+	//sample.GetValue()
+	return sample.GetTimestamp(), sample.GetValue()
+}
+
+// this is adapted from Go 1.15 for older versions
+// TODO: Reuse upstream once that version is common
+func readactedURL(u *url.URL) string {
```
Comment (replay) [inline pkg/commands/remote_read.go:166]:
```suggestion
func redactedURL(u *url.URL) string {
```

File: pkg/commands/remote_read.go [hunk 4]
```diff
@@ -0,0 +1,441 @@
...
+		if !ok {
+			return nil, fmt.Errorf("unexpected type %T", readClient)
+		}
+		client.Client.Transport = &setTenantIDTransport{
+			RoundTripper: client.Client.Transport,
+			tenantID:     c.tenantID,
+		}
+	}
+
+	log.Infof("Create remote read client using endpoint '%s'", readactedURL(addressURL))
```
Comment (replay) [inline pkg/commands/remote_read.go:217]:
```suggestion
	log.Infof("Created remote read client using endpoint '%s'", readactedURL(addressURL))
```

File: pkg/commands/remote_read.go [hunk 5]
```diff
@@ -0,0 +1,441 @@
...
+
+	if c.tsdbPath == "" {
+		c.tsdbPath, err = ioutil.TempDir("", "cortextool-tsdb")
+		if err != nil {
+			return err
+		}
+		log.Infof("Created TSDB in path '%s'", c.tsdbPath)
+	} else {
+		if _, err := os.Stat(c.tsdbPath); err != nil && os.IsNotExist(err) {
+			if err := os.Mkdir(c.tsdbPath, 0755); err != nil {
```
Comment (replay) [inline pkg/commands/remote_read.go:395]:
```suggestion
			if err = os.Mkdir(c.tsdbPath, 0755); err != nil {
```

File: pkg/commands/remote_read.go [hunk 6]
```diff
@@ -0,0 +1,441 @@
...
+		}
+		log.Infof("Using existing TSDB in path '%s'", c.tsdbPath)
+	}
+
+	mint := model.TimeFromUnixNano(from.UnixNano())
+	maxt := model.TimeFromUnixNano(to.UnixNano())
+
+	timeseries, err := query(context.Background())
+	if err != nil {
+		return nil
```
Comment (replay) [inline pkg/commands/remote_read.go:408]:
```suggestion
		return err
```

File: pkg/backfill/backfill.go (hunk 1 above)
Comment (simonswine) [reply pkg/backfill/backfill.go:83]: I first came to contact with it, when it was suggested in a code review. To be honest it blew my mind. The backfill code is an adapted fork from https://github.com/prometheus/prometheus/blob/master/cmd/promtool/backfill.go

File: pkg/commands/remote_read.go (hunk 2 above)
Comment (simonswine) [reply pkg/commands/remote_read.go]: Good catch :+1:

**Review by replay — APPROVED** [review]
LGTM

**Review by jtlisi — APPROVED** [review]
LGTM

**Pull request merged**